### ISSUE TITLE
node listener not synced - log flood

### ISSUE THREAD

**Issue by nickma82** [openshift/cluster-etcd-operator#466]
I'm having a log flood (600 logs/min) with
quay.io/openshift-release-dev/ocp-v4.0-art-dev
which is auto deployed.

```
OPERATOR_IMAGE_VERSION | 4.5.13
cmd:     cluster-etcd-operator operator
args:     --config=/var/run/configmaps/config/config.yaml -v=4
```

There's a 
* `"configmap/etcd-pod": missing env var values` and also a `"EtcdMembersController" controller failed to sync "key", err: node lister not synced` maybe something with them?
* maybe it's set to `-v=4` in the args because of the dev version - how to reduce it?
Any idea?

Log:
```I1016 06:30:25.708797       1 base_controller.go:54] Starting #1 worker of StatusSyncer_etcd controller ...
I1016 06:30:25.709796       1 status_controller.go:172] clusteroperator/etcd diff {"status":{"conditions":[{"lastTransitionTime":"2020-08-31T11:25:54Z","message":"NodeControllerDegraded: All master nodes are ready\nEtcdMembersControllerDegraded: node lister not synced\nEtcdMembersDegraded: No unhealthy members found","reason":"AsExpected","status":"False","type":"Degraded"},{"lastTransitionTime":"2020-10-08T08:30:05Z","message":"NodeInstallerProgressing: 3 nodes are at revision 7\nEtcdMembersProgressing: No unstarted etcd members found","reason":"AsExpected","status":"False","type":"Progressing"},{"lastTransitionTime":"2020-07-13T09:36:06Z","message":"StaticPodsAvailable: 3 nodes are active; 3 nodes are at revision 7\nEtcdMembersAvailable: 3 members are available","reason":"AsExpected","status":"True","type":"Available"},{"lastTransitionTime":"2020-07-10T12:38:42Z","reason":"AsExpected","status":"True","type":"Upgradeable"}]}}
I1016 06:30:25.709936       1 shared_informer.go:230] Caches are synced for UnsupportedConfigOverridesController 
I1016 06:30:25.709955       1 base_controller.go:54] Starting #1 worker of UnsupportedConfigOverridesController controller ...
I1016 06:30:25.710074       1 shared_informer.go:230] Caches are synced for PruneController 
I1016 06:30:25.710082       1 base_controller.go:54] Starting #1 worker of PruneController controller ...
I1016 06:30:25.710294       1 shared_informer.go:230] Caches are synced for NodeController 
I1016 06:30:25.710314       1 base_controller.go:54] Starting #1 worker of NodeController controller ...
I1016 06:30:25.710405       1 prune_controller.go:134] no revision IDs currently eligible to prune
I1016 06:30:25.710413       1 prune_controller.go:286] No excluded revisions to prune, skipping
E1016 06:30:25.717733       1 base_controller.go:180] "EtcdMembersController" controller failed to sync "key", err: node lister not synced
E1016 06:30:25.721427       1 base_controller.go:180] "EtcdMemberIPMigrator" controller failed to sync "key", err: node lister not synced
I1016 06:30:25.722166       1 prune_controller.go:134] no revision IDs currently eligible to prune
I1016 06:30:25.722200       1 prune_controller.go:286] No excluded revisions to prune, skipping
E1016 06:30:25.723058       1 base_controller.go:180] "EtcdMemberIPMigrator" controller failed to sync "key", err: node lister not synced
E1016 06:30:25.724033       1 base_controller.go:180] "EtcdMemberIPMigrator" controller failed to sync "key", err: node lister not synced
I1016 06:30:25.723132       1 event.go:278] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-etcd-operator", Name:"etcd-operator", UID:"c6510a17-c9fa-4d0e-914f-c4e9339d3f53", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'OperatorStatusChanged' Status for clusteroperator/etcd changed: Degraded message changed from "NodeControllerDegraded: All master nodes are ready\nEtcdMembersDegraded: No unhealthy members found" to "NodeControllerDegraded: All master nodes are ready\nEtcdMembersControllerDegraded: node lister not synced\nEtcdMembersDegraded: No unhealthy members found"
I1016 06:30:25.723715       1 status_controller.go:172] clusteroperator/etcd diff {"status":{"conditions":[{"lastTransitionTime":"2020-08-31T11:25:54Z","message":"NodeControllerDegraded: All master nodes are ready\nEtcdMembersControllerDegraded: node lister not synced\nEtcdMemberIPMigratorDegraded: node lister not synced\nEtcdMembersDegraded: No unhealthy members found","reason":"AsExpected","status":"False","type":"Degraded"},{"lastTransitionTime":"2020-10-08T08:30:05Z","message":"NodeInstallerProgressing: 3 nodes are at revision 7\nEtcdMembersProgressing: No unstarted etcd members found","reason":"AsExpected","status":"False","type":"Progressing"},{"lastTransitionTime":"2020-07-13T09:36:06Z","message":"StaticPodsAvailable: 3 nodes are active; 3 nodes are at revision 7\nEtcdMembersAvailable: 3 members are available","reason":"AsExpected","status":"True","type":"Available"},{"lastTransitionTime":"2020-07-10T12:38:42Z","reason":"AsExpected","status":"True","type":"Upgradeable"}]}}
E1016 06:30:25.726724       1 base_controller.go:180] "EtcdMemberIPMigrator" controller failed to sync "key", err: node lister not synced
I1016 06:30:25.729423       1 event.go:278] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-etcd-operator", Name:"etcd-operator", UID:"c6510a17-c9fa-4d0e-914f-c4e9339d3f53", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'OperatorStatusChanged' Status for clusteroperator/etcd changed: Degraded message changed from "NodeControllerDegraded: All master nodes are ready\nEtcdMembersControllerDegraded: node lister not synced\nEtcdMembersDegraded: No unhealthy members found" to "NodeControllerDegraded: All master nodes are ready\nEtcdMembersControllerDegraded: node lister not synced\nEtcdMemberIPMigratorDegraded: node lister not synced\nEtcdMembersDegraded: No unhealthy members found"
I1016 06:30:25.729858       1 status_controller.go:172] clusteroperator/etcd diff {"status":{"conditions":[{"lastTransitionTime":"2020-08-31T11:25:54Z","message":"NodeControllerDegraded: All master nodes are ready\nEtcdMembersControllerDegraded: node lister not synced\nEtcdMemberIPMigratorDegraded: node lister not synced\nEtcdMembersDegraded: No unhealthy members found","reason":"AsExpected","status":"False","type":"Degraded"},{"lastTransitionTime":"2020-10-08T08:30:05Z","message":"NodeInstallerProgressing: 3 nodes are at revision 7\nEtcdMembersProgressing: No unstarted etcd members found","reason":"AsExpected","status":"False","type":"Progressing"},{"lastTransitionTime":"2020-07-13T09:36:06Z","message":"StaticPodsAvailable: 3 nodes are active; 3 nodes are at revision 7\nEtcdMembersAvailable: 3 members are available","reason":"AsExpected","status":"True","type":"Available"},{"lastTransitionTime":"2020-07-10T12:38:42Z","reason":"AsExpected","status":"True","type":"Upgradeable"}]}}
I1016 06:30:25.736576       1 event.go:278] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-etcd-operator", Name:"etcd-operator", UID:"c6510a17-c9fa-4d0e-914f-c4e9339d3f53", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'OperatorStatusChanged' Status for clusteroperator/etcd changed: Degraded message changed from "NodeControllerDegraded: All master nodes are ready\nEtcdMembersControllerDegraded: node lister not synced\nEtcdMembersDegraded: No unhealthy members found" to "NodeControllerDegraded: All master nodes are ready\nEtcdMembersControllerDegraded: node lister not synced\nEtcdMemberIPMigratorDegraded: node lister not synced\nEtcdMembersDegraded: No unhealthy members found"
E1016 06:30:25.743531       1 base_controller.go:180] "BootstrapTeardownController" controller failed to sync "key", err: node lister not synced
I1016 06:30:25.743591       1 prune_controller.go:134] no revision IDs currently eligible to prune
I1016 06:30:25.743603       1 prune_controller.go:286] No excluded revisions to prune, skipping
E1016 06:30:25.743676       1 base_controller.go:180] "EtcdMemberIPMigrator" controller failed to sync "key", err: node lister not synced
E1016 06:30:25.743758       1 base_controller.go:180] "BootstrapTeardownController" controller failed to sync "key", err: node lister not synced
E1016 06:30:25.744835       1 base_controller.go:180] "EtcdMemberIPMigrator" controller failed to sync "key", err: node lister not synced
I1016 06:30:25.745898       1 status_controller.go:172] clusteroperator/etcd diff {"status":{"conditions":[{"lastTransitionTime":"2020-08-31T11:25:54Z","message":"NodeControllerDegraded: All master nodes are ready\nEtcdMembersControllerDegraded: node lister not synced\nEtcdMemberIPMigratorDegraded: node lister not synced\nBootstrapTeardownDegraded: node lister not synced\nEtcdMembersDegraded: No unhealthy members found","reason":"AsExpected","status":"False","type":"Degraded"},{"lastTransitionTime":"2020-10-08T08:30:05Z","message":"NodeInstallerProgressing: 3 nodes are at revision 7\nEtcdMembersProgressing: No unstarted etcd members found","reason":"AsExpected","status":"False","type":"Progressing"},{"lastTransitionTime":"2020-07-13T09:36:06Z","message":"StaticPodsAvailable: 3 nodes are active; 3 nodes are at revision 7\nEtcdMembersAvailable: 3 members are available","reason":"AsExpected","status":"True","type":"Available"},{"lastTransitionTime":"2020-07-10T12:38:42Z","reason":"AsExpected","status":"True","type":"Upgradeable"}]}}
E1016 06:30:25.747094       1 base_controller.go:180] "BootstrapTeardownController" controller failed to sync "key", err: node lister not synced
E1016 06:30:25.748831       1 base_controller.go:180] "BootstrapTeardownController" controller failed to sync "key", err: node lister not synced
I1016 06:30:25.751607       1 event.go:278] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-etcd-operator", Name:"etcd-operator", UID:"c6510a17-c9fa-4d0e-914f-c4e9339d3f53", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'OperatorStatusChanged' Status for clusteroperator/etcd changed: Degraded message changed from "NodeControllerDegraded: All master nodes are ready\nEtcdMembersControllerDegraded: node lister not synced\nEtcdMemberIPMigratorDegraded: node lister not synced\nEtcdMembersDegraded: No unhealthy members found" to "NodeControllerDegraded: All master nodes are ready\nEtcdMembersControllerDegraded: node lister not synced\nEtcdMemberIPMigratorDegraded: node lister not synced\nBootstrapTeardownDegraded: node lister not synced\nEtcdMembersDegraded: No unhealthy members found"
E1016 06:30:25.767014       1 base_controller.go:180] "EtcdMemberIPMigrator" controller failed to sync "key", err: node lister not synced
E1016 06:30:25.789268       1 base_controller.go:180] "BootstrapTeardownController" controller failed to sync "key", err: node lister not synced
E1016 06:30:25.798286       1 base_controller.go:180] "EtcdMembersController" controller failed to sync "key", err: node lister not synced
I1016 06:30:25.870112       1 client.go:360] parsed scheme: "endpoint"
I1016 06:30:25.870202       1 endpoint.go:68] ccResolverWrapper: sending new addresses to cc: [{https://10.197.1.103:2379  <nil> 0 <nil>} {https://10.197.1.101:2379  <nil> 0 <nil>} {https://10.197.1.102:2379  <nil> 0 <nil>}]
I1016 06:30:25.888751       1 client.go:360] parsed scheme: "endpoint"
I1016 06:30:25.888837       1 endpoint.go:68] ccResolverWrapper: sending new addresses to cc: [{https://10.197.1.102:2379  <nil> 0 <nil>} {https://10.197.1.103:2379  <nil> 0 <nil>} {https://10.197.1.101:2379  <nil> 0 <nil>}]
I1016 06:30:25.889661       1 status_controller.go:172] clusteroperator/etcd diff {"status":{"conditions":[{"lastTransitionTime":"2020-08-31T11:25:54Z","message":"NodeControllerDegraded: All master nodes are ready\nEtcdMembersControllerDegraded: node lister not synced\nEtcdMemberIPMigratorDegraded: node lister not synced\nEtcdMembersDegraded: No unhealthy members found","reason":"AsExpected","status":"False","type":"Degraded"},{"lastTransitionTime":"2020-10-08T08:30:05Z","message":"NodeInstallerProgressing: 3 nodes are at revision 7\nEtcdMembersProgressing: No unstarted etcd members found","reason":"AsExpected","status":"False","type":"Progressing"},{"lastTransitionTime":"2020-07-13T09:36:06Z","message":"StaticPodsAvailable: 3 nodes are active; 3 nodes are at revision 7\nEtcdMembersAvailable: 3 members are available","reason":"AsExpected","status":"True","type":"Available"},{"lastTransitionTime":"2020-07-10T12:38:42Z","reason":"AsExpected","status":"True","type":"Upgradeable"}]}}
I1016 06:30:25.898067       1 client.go:360] parsed scheme: "endpoint"
I1016 06:30:25.898095       1 endpoint.go:68] ccResolverWrapper: sending new addresses to cc: [{https://10.197.1.102:2379  <nil> 0 <nil>} {https://10.197.1.101:2379  <nil> 0 <nil>} {https://10.197.1.103:2379  <nil> 0 <nil>}]
I1016 06:30:25.898232       1 event.go:278] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-etcd-operator", Name:"etcd-operator", UID:"c6510a17-c9fa-4d0e-914f-c4e9339d3f53", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'OperatorStatusChanged' Status for clusteroperator/etcd changed: Degraded message changed from "NodeControllerDegraded: All master nodes are ready\nEtcdMembersControllerDegraded: node lister not synced\nEtcdMemberIPMigratorDegraded: node lister not synced\nBootstrapTeardownDegraded: node lister not synced\nEtcdMembersDegraded: No unhealthy members found" to "NodeControllerDegraded: All master nodes are ready\nEtcdMembersControllerDegraded: node lister not synced\nEtcdMemberIPMigratorDegraded: node lister not synced\nEtcdMembersDegraded: No unhealthy members found"
I1016 06:30:25.898386       1 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I1016 06:30:25.898454       1 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I1016 06:30:25.898461       1 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I1016 06:30:25.899308       1 client.go:360] parsed scheme: "endpoint"
I1016 06:30:25.899754       1 endpoint.go:68] ccResolverWrapper: sending new addresses to cc: [{https://10.197.1.102:2379  <nil> 0 <nil>}]
I1016 06:30:25.899436       1 client.go:360] parsed scheme: "endpoint"
I1016 06:30:25.899845       1 endpoint.go:68] ccResolverWrapper: sending new addresses to cc: [{https://10.197.1.101:2379  <nil> 0 <nil>}]
I1016 06:30:25.899569       1 client.go:360] parsed scheme: "endpoint"
I1016 06:30:25.899957       1 endpoint.go:68] ccResolverWrapper: sending new addresses to cc: [{https://10.197.1.103:2379  <nil> 0 <nil>}]
I1016 06:30:25.906263       1 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I1016 06:30:25.906319       1 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I1016 06:30:25.906450       1 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I1016 06:30:25.908453       1 shared_informer.go:230] Caches are synced for EtcdEndpointsController 
I1016 06:30:25.908482       1 base_controller.go:54] Starting #1 worker of EtcdEndpointsController controller ...
I1016 06:30:25.910229       1 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I1016 06:30:25.911120       1 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I1016 06:30:25.911406       1 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I1016 06:30:25.911725       1 client.go:360] parsed scheme: "endpoint"
I1016 06:30:25.911746       1 endpoint.go:68] ccResolverWrapper: sending new addresses to cc: [{https://10.197.1.103:2379  <nil> 0 <nil>} {https://10.197.1.101:2379  <nil> 0 <nil>} {https://10.197.1.102:2379  <nil> 0 <nil>}]
I1016 06:30:25.916686       1 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I1016 06:30:25.916754       1 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I1016 06:30:25.916803       1 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I1016 06:30:25.928858       1 client.go:360] parsed scheme: "endpoint"
....
I1016 06:30:26.308749       1 shared_informer.go:230] Caches are synced for ScriptController 
I1016 06:30:26.308759       1 base_controller.go:54] Starting #1 worker of ScriptController controller ...
E1016 06:30:26.308832       1 base_controller.go:180] "TargetConfigController" controller failed to sync "key", err: missing env var values
E1016 06:30:26.318235       1 base_controller.go:180] "ScriptController" controller failed to sync "key", err: "configmap/etcd-pod": missing env var values
I1016 06:30:26.319233       1 client.go:360] parsed scheme: "endpoint"
I1016 06:30:26.319291       1 endpoint.go:68] ccResolverWrapper: sending new addresses to cc: [{https://10.197.1.102:2379  <nil> 0 <nil>} {https://10.197.1.103:2379  <nil> 0 <nil>} {https://10.197.1.101:2379  <nil> 0 <nil>}]
I1016 06:30:26.320248       1 status_controller.go:172] clusteroperator/etcd diff {"status":{"conditions":[{"lastTransitionTime":"2020-08-31T11:25:54Z","message":"NodeControllerDegraded: All master nodes are ready\nScriptControllerDegraded: \"configmap/etcd-pod\": missing env var values\nEtcdMembersDegraded: No unhealthy members found","reason":"AsExpected","status":"False","type":"Degraded"},{"lastTransitionTime":"2020-10-08T08:30:05Z","message":"NodeInstallerProgressing: 3 nodes are at revision 7\nEtcdMembersProgressing: No unstarted etcd members found","reason":"AsExpected","status":"False","type":"Progressing"},{"lastTransitionTime":"2020-07-13T09:36:06Z","message":"StaticPodsAvailable: 3 nodes are active; 3 nodes are at revision 7\nEtcdMembersAvailable: 3 members are available","reason":"AsExpected","status":"True","type":"Available"},{"lastTransitionTime":"2020-07-10T12:38:42Z","reason":"AsExpected","status":"True","type":"Upgradeable"}]}}
I1016 06:30:26.326908       1 event.go:278] Event(v1.ObjectReference{Kind:"Deployment", Namespace:"openshift-etcd-operator", Name:"etcd-operator", UID:"c6510a17-c9fa-4d0e-914f-c4e9339d3f53", APIVersion:"apps/v1", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'OperatorStatusChanged' Status for clusteroperator/etcd changed: Degraded message changed from "NodeControllerDegraded: All master nodes are ready\nEtcdMembersDegraded: No unhealthy members found" to "NodeControllerDegraded: All master nodes are ready\nScriptControllerDegraded: \"configmap/etcd-pod\": missing env var values\nEtcdMembersDegraded: No unhealthy members found"
I1016 06:30:26.331141       1 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I1016 06:30:26.331252       1 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
I1016 06:30:26.331295       1 controlbuf.go:508] transport: loopyWriter.run returning. connection error: desc = "transport is closing"
```

**Comment by hexfusion** [comment]
@nickma82 thanks for the report

@retroflexer can you take a look, please.

**Comment by hexfusion** [comment]
closed via #389
